apps/E2E/src/common/BasePage.ts

```diff
@@ -72,6 +72,31 @@ export class BasePage {
   /******************************************************************/
   /**************** UI Element Interaction Methods ******************/
   /******************************************************************/
+
+  /**
+   * Checks to see if an element attribute is strictly equal to an expected value the user passes in.
+   * The advantage to this over testing using .isEqual in a spec is that this throws a detailed error if
+   * the expected and actual values don't match.
+   * This should never be called in a spec. Other PageObjects should use these to make accessibility / attribute testing easier for them. */
+  protected async _attributeIsEqualTo(element: WebdriverIO.Element, attribute: Attribute, expectedValue: any): Promise<boolean> {
+    const actualValue = await element.getAttribute(attribute);
+    if (expectedValue !== actualValue) {
+      throw new Error(
+        `On ${this._pageName}, a test component should have attribute, ${attributeToEnumName[attribute]} (${attribute}), equal to ${expectedValue}.
+        Instead, ${attributeToEnumName[attribute]} is equal to ${actualValue}.`,
+      );
+    }
+    return true;
+  }
+
+  async accessibilityLabelIsEqualTo(element: Promise<WebdriverIO.Element>, expectedValue: any): Promise<boolean> {
```

Comment:
I think we should consolidate these two functions even further. For "x" amount of properties, we shouldn't make "x" amount of methods...

For example, following this pattern, we'd have a `accessibilityLabelIsEqualTo()`, `accessibilityRoleIsEqualTo()`, `expandCollapsestateEqualTo()`, `IsTogglePatternEqualTo()`, etc...

Instead, let's have one method that takes in a Property Name to query for, and the element to query it from.

Such as,

`CompareAttribute(element: Promise<WebDriverIO.Element>, attribute: Attribute, expectedValue: any)`.

Then, in the specs, we can use:
`ButtonExperimentalPageObject.CompareAttribute(ButtonExperimentalPageObject._primaryComponent, Attribute.AccessibilityRole, BUTTON_A11Y_ROLE)`

Reply:
And `CompareAttribute()` body would be:
`return this._attributeIsEqualTo(await element, attribute, exectedValue)`

Reply:
Or, we can even re-name `_attributeIsEqualTo` to the `CompareAttribute` and simply use that in spec files